app/concepcts/cinema_halls/use_cases/update.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module CinemaHalls
+  module UseCases
+    class Update
+      def initialize(id:, params:, repository: CinemaHallRepository.new)
+        @id = id.to_i
+        @repository = repository
+        @params = params
+      end
+
+      def call
+        raise "Cinema Hall with #{id} don't exist" if repository.find(id).present? == false
```

Comment:
@wmaciejak

@lilka You can create your own custom error that inherits from RecordNotFound. It can sound like "CinemaHallNotFoundError". 
You might add errors folder and add class there.

./errors/cinema_hall_not_found.rb

'module CinemaHalls
     module Errors
        class CinemaHallNotFound < ActiveRecord::RecordNotFound end
     end
 end`
 
 and do The same Wojtek mentioned.